src/Altinn.Platform/Altinn.Platform.Events/Events/Repository/EventsRepository.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using Altinn.Platform.Events.Configuration;
+using Altinn.Platform.Events.Models;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.Platform.Events.Repository
+{
+    /// <summary>
+    /// Handles events repository. Notice that the all methods should modify the Subject attribute of the
+    /// CloudEvent, since cosmosDb fails if Subject contains slashes '/'.
+    /// </summary>
+    public class EventsRepository : IEventsRepository
+    {
+        private readonly ILogger _logger;
+        private readonly Uri _collectionUri;
+        private readonly string _collectionId = "events";
+        private readonly string _partitionKey = "/subject";
+        private readonly string _triggerId = "trgUpdateItemTimestamp";
+        private readonly DocumentClient _client;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventsRepository"/> class.
+        /// </summary>
+        /// <param name="cosmosettings">the configuration settings for cosmos database</param>
+        /// <param name="logger">dependency injection of logger</param>
+        public EventsRepository(IOptions<AzureCosmosSettings> cosmosettings, ILogger<EventsRepository> logger)
+        {
+            this._logger = logger;
+
+            var database = new CosmosDatabaseHandler(cosmosettings.Value);
```

Comment:
no biggie, but we try to avoid using var wherever we know the explicit type

Reply:
Fixed :)